### PR TITLE
fix: set 0 min gas prices

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

This is necessary for testing since keplr's suggestchain feature only allows default gas prices computations. We should add this in another prop after v3

This is initially opened for discussion

**Update:** Let's use this to build a network to test keplr against for the frontend team.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by robert-zaremba** [comment]
how about the following:
1. for firedrill we leave the gas prices update
2. for beta testing we create a new chain, based on this branch (so instead of merging this branch, we will create a new binary from this branch).

File: ante/fee.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func checkFees(minGasPrices sdk.DecCoins, fees sdk.Coins, gasLimit uint64) error
 		requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
 	}
 
-	if !fees.IsAnyGTE(requiredFees) {
+	if requiredFees.IsAnyGT(fees) {
```
Comment (toteki) [inline ante/fee.go:66]: From docs:

> NOTE: IsAnyGTE operates under the invariant that both coin sets are sorted by denominations and there exists no zero coins.

When minimum fee is zero, there are zero coins.

File: ante/fee.go [hunk 2]
```diff
@@ -63,7 +63,10 @@ func checkFees(minGasPrices sdk.DecCoins, fees sdk.Coins, gasLimit uint64) error
 		requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
 	}
 
-	if !fees.IsAnyGTE(requiredFees) {
+	// Clear any zero coins from requiredFees in the case of zero min gas price
+	requiredFees = sdk.NewCoins(requiredFees...)
+
+	if !requiredFees.Empty() && !fees.IsAnyGTE(requiredFees) {
```
Comment (toteki) [inline ante/fee.go:69]: From docs:

> IsAnyGTE operates under the invariant that both coin sets are sorted by denominations and there exists no zero coins.

File: ante/fee.go [hunk 3]
```diff
@@ -63,7 +63,10 @@ func checkFees(minGasPrices sdk.DecCoins, fees sdk.Coins, gasLimit uint64) error
 		requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
 	}
 
-	if !fees.IsAnyGTE(requiredFees) {
+	// Clear any zero coins from requiredFees in the case of zero min gas price
+	requiredFees = sdk.NewCoins(requiredFees...)
```
Comment (robert-zaremba) [inline ante/fee.go:67]: this line is not needed - let's update the loop above:
```
	requiredFees := make(sdk.Coins, len(minGasPrices), 0)

	// Determine the required fees by multiplying each required minimum gas
	// price by the gas limit, where fee = ceil(minGasPrice * gasLimit).
	glDec := sdk.NewDec(int64(gasLimit))
	for _, gp := range minGasPrices {
        if gp.IsZero() || gasLimit == 0 {
           continue
        }
		fee := gp.Amount.Mul(glDec)
		requiredFees = append(requiredFees, sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt()))
	}
```

**Review by toteki — APPROVED** [review]

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**